libraries/stdlib/genglsl/mx_heighttonormal_vector3.glsl

```diff
@@ -0,0 +1,32 @@
+void mx_heighttonormal_vector3(float height, float scale, vec2 texcoord, out vec3 result)
+{
+    // Compute the gradient of the heightfield with respect to the screen.
+    vec2 dHdS = vec2(dFdx(height), dFdy(height)) * scale;
+
+    // Compute the gradient of the texture coordinates with respect to the screen.
+    vec2 dUdS = vec2(dFdx(texcoord.x), dFdy(texcoord.x));
+    vec2 dVdS = vec2(dFdx(texcoord.y), dFdy(texcoord.y));
+
+    // Use the chain rule to compute the gradient of the heightfield with
+    // respect to the texture coordinates.
+    vec2 dHdT;
+    float det = dUdS.x * dVdS.y - dUdS.y * dVdS.x;
+    if (abs(det) > M_FLOAT_EPS)
+    {
+        mat2 invJacobian = mat2(-dVdS.y, dUdS.y, dVdS.x, -dUdS.x) / det;
+        dHdT = invJacobian * dHdS;
+    }
+    else
+    {
+        dHdT = vec2(0.0);
+    }
+
+    // Scale the results for parity with traditional Sobel filtering.
+    // https://nrsyed.com/2018/02/18/edge-detection-in-images-how-to-derive-the-sobel-operator/
+    const float SOBEL_SCALE_FACTOR = 1.0 / 16.0;
+    dHdT *= SOBEL_SCALE_FACTOR;
+
+    // Convert the gradient to a normal and encode for storage.
+    vec3 n = normalize(vec3(dHdT.x, dHdT.y, 1.0));
```

Comment:
It seems the normal is inverted (pointing inward) and you may have issues with numeric instability if dHdT is very large (Z map become very small). Suggested:
```
 float z_scale = max(1.0, length(dHdT));
 vec3 n = normalize(vec3(-dHdT.x, -dHdT.y, z_scale));
 ```

Reply:
Strange. Based on the input image, this is what I currently get in MaterialXView which is pointing out from the sphere (using sphere.obj from the resources folder).

![image](https://github.com/user-attachments/assets/25e72c97-2d24-4abe-ae6c-ab6a345e0caa)

Reply:
Ok, here is a comparison of the GLSL output of the Render Test Suite, with the first image being the original Sobel filtering, and the second image being the new Chain Rule implementation:

GLSL Sobel:
![HeightToNormal_GLSL_Sobel](https://github.com/user-attachments/assets/ebdab34d-f29d-4629-aa59-ece7b6b83d11)

GLSL Chain Rule (current PR)
![HeightToNormal_GLSL_ChainRule](https://github.com/user-attachments/assets/26e23fff-e5a4-4ffe-8a37-011cef08c101)

Reply:
To my eye, this confirms that the orientation of the normal map is the preserved by the new implementation, and to my eye, it provides better handling of edge cases where the tiling image is not aligned with the screen plane.

In this same render test run, I'm seeing some stretching issues in OSL that are likely caused by one of my recent optimizations, so I'll fix this up before posting a full PDF of the render comparisons.

Reply:
What confused me visually to make me "see" the normals as facing "in" vs "out" was that  "up" orientation differs for the MaterialXViewer vs unit tests. Sorry for the mistaken "noise".
It would be good to address the image flip inconsistency at some point.